### DISCUSSION TITLE
Add us army to config OPFOR

### DISCUSSION THREAD

**Discussion by Pullgung** [KillahPotatoes/KP-Liberation#907] [General]
Can add US army to config in OPFOR??

**Comment by H4yn3x** [comment]
Not sure if it is useful, but I've done it through custom OPFOR. Do note that the below demands RHS USAF and makes it so the soldiers wear marpat cammo.

`/*
    Needed Mods:
    - None

    Optional Mods:
    - None
*/

// Enemy infantry classes
opfor_officer = "rhsusf_usmc_marpat_wd_officer";                                          // Officer
opfor_squad_leader = "rhsusf_usmc_marpat_wd_squadleader";                                  // Squad Leader
opfor_team_leader = "rhsusf_usmc_marpat_wd_teamleader";                                   // Team Leader
opfor_sentry = "rhsusf_usmc_marpat_wd_rifleman_light";                                      // Rifleman (Lite)
opfor_rifleman = "rhsusf_usmc_marpat_wd_rifleman_m4";                                         // Rifleman
opfor_rpg = "rhsusf_usmc_marpat_wd_smaw";                                          // Rifleman (LAT)
opfor_grenadier = "rhsusf_usmc_marpat_wd_grenadier";                                     // Grenadier
opfor_machinegunner = "rhsusf_usmc_marpat_wd_autorifleman_m249";                                 // Autorifleman
opfor_heavygunner = "rhsusf_usmc_marpat_wd_machinegunner";                                  // Heavy Gunner
opfor_marksman = "rhsusf_usmc_marpat_wd_marksman";                                       // Marksman
opfor_sharpshooter = "rhsusf_usmc_marpat_wd_sniper_m110";                                // Sharpshooter
opfor_sniper = "rhsusf_usmc_marpat_wd_sniper_M107";                                            // Sniper
opfor_at = "rhsusf_usmc_marpat_wd_javelin";                                            // AT Specialist
opfor_aa = "rhsusf_usmc_marpat_wd_stinger";                                            // AA Specialist
opfor_medic = "rhsusf_navy_marpat_wd_medic";                                              // Combat Life Saver
opfor_engineer = "rhsusf_usmc_marpat_wd_engineer";                                        // Engineer
opfor_paratrooper = "rhsusf_army_ocp_rifleman_101st";                                   // Paratrooper

// Enemy vehicles used by secondary objectives.
opfor_mrap = "rhsusf_m1240a1_usmc_wd";                                             // Ifrit
opfor_mrap_armed = "rhsusf_m1240a1_m2_usmc_wd";                                   // Ifrit (HMG)
opfor_transport_helo = "RHS_UH1Y_UNARMED";                   // Mi-290 Taru (Bench)
opfor_transport_truck = "rhsusf_M1078A1P2_WD_fmtv_usarmy";                         // Tempest Transport (Covered)
opfor_ammobox_transport = "rhsusf_M1078A1P2_WD_flatbed_fmtv_usarmy";                     // Tempest Transport (Open) -> Has to be able to transport resource crates!
opfor_fuel_truck = "rhsusf_M978A4_BKIT_usarmy_wd";                                 // Tempest Fuel
opfor_ammo_truck = "rhsusf_M977A4_AMMO_BKIT_usarmy_wd";                                 // Tempest Ammo
opfor_fuel_container = "Land_Pod_Heli_Transport_04_fuel_F";             // Taru Fuel Pod
opfor_ammo_container = "Land_Pod_Heli_Transport_04_ammo_F";             // Taru Ammo Pod
opfor_flag = "FlagCarrierNATO_EP1";                                             // Flag

/* Adding a value to these arrays below will add them to a one out of however many in the array, random pick chance.
Therefore, adding the same value twice or three times means they are more likely to be chosen more often. */

/* Militia infantry. Lightweight soldier classnames the game will pick from randomly as sector defenders.
Think of them like garrison or military police forces, which are more meant to control the local population instead of fighting enemy armies. */
militia_squad = [
    "rhsusf_usmc_marpat_wd_rifleman_light",                                                     // Rifleman (Lite)
    "rhsusf_usmc_marpat_wd_rifleman_light",                                                     // Rifleman (Lite)
    "rhsusf_usmc_marpat_wd_rifleman",                                                     // Rifleman
    "rhsusf_usmc_marpat_wd_rifleman_m4",                                                     // Rifleman
    "rhsusf_usmc_marpat_wd_smaw",                                                 // Rifleman (AT)
    "rhsusf_usmc_marpat_wd_autorifleman_m249",                                                  // Autorifleman
    "rhsusf_usmc_marpat_wd_marksman",                                                   // Marksman
    "rhsusf_navy_marpat_wd_medic",                                               // Medic
    "rhsusf_usmc_marpat_wd_engineer"                                                    // Engineer
];

// Militia vehicles. Lightweight vehicle classnames the game will pick from randomly as sector defenders. Can also be empty for only infantry milita.
militia_vehicles = [
    "rhsusf_m1025_w_s_m2"                                                  // Qilin (armed)
];

// All enemy vehicles that can spawn as sector defenders and patrols at high enemy combat readiness (aggression levels).
opfor_vehicles = [
    "rhsusf_m1240a1_m2_usmc_wd",                                                  // Ifrit (HMG)
    "rhsusf_m1240a1_m2_usmc_wd",                                                  // Ifrit (GMG)
    "rhsusf_m1025_w_s_Mk19",                                                  // Ifrit (GMG)
    "rhsusf_m1045_w_s",                                                    // Qilin (AT)
    "rhsusf_m113_usarmy",                                        // BTR-K Kamysh
    "rhsusf_m113_usarmy",                                        // BTR-K Kamysh
    "RHS_M2A2_wd",                                            // ZSU-39 Tigris
    "RHS_M2A2_wd",                                            // ZSU-39 Tigris
    "rhsusf_m1a1aimwd_usarmy",                                                // T-100 Varsuk
    "rhsusf_m1a1aimwd_usarmy",                                                // T-100 Varsuk
    "rhsusf_m1a2sep2wd_usarmy",                                                // T-140 Angara
    "rhsusf_m1a2sep2wd_usarmy"                                                // T-140K Angara
];

// All enemy vehicles that can spawn as sector defenders and patrols but at a lower enemy combat readiness (aggression levels).
opfor_vehicles_low_intensity = [
    "rhsusf_m1240a1_m2_usmc_wd",                                                  // Ifrit (HMG)
    "rhsusf_m1240a1_m2_usmc_wd",                                                  // Ifrit (HMG)
    "rhsusf_m1045_w_s",                                                    // Qilin (AT)
    "rhsusf_stryker_m1126_m2_wd",                                          // MSE-3 Marid
    "rhsusf_m113_usarmy"                                         // BTR-K Kamysh
];

// All enemy vehicles that can spawn as battlegroups, either assaulting or as reinforcements, at high enemy combat readiness (aggression levels).
opfor_battlegroup_vehicles = [
    "rhsusf_m1240a1_m2_usmc_wd",                                                  // Ifrit (HMG)
    "rhsusf_m1025_w_s_Mk19",                                                  // Ifrit (GMG)
    "rhsusf_m1025_w_s_Mk19",                                                  // Ifrit (GMG)
    "rhsusf_m1045_w_s",                                                    // Qilin (AT)
    "rhsusf_M1078A1P2_WD_fmtv_usarmy",                                           // Tempest Transport
    "rhsusf_M1078A1P2_WD_fmtv_usarmy",                                             // Tempest Transport (Covered)
    "rhsusf_m113_usarmy",                                        // BTR-K Kamysh
    "rhsusf_m113_usarmy",                                        // BTR-K Kamysh
    "RHS_M2A2_wd",                                            // ZSU-39 Tigris
    "RHS_M2A2_wd",                                            // ZSU-39 Tigris
    "rhsusf_m1a1aimwd_usarmy",                                                // T-100 Varsuk
    "rhsusf_m1a1aimwd_usarmy",                                                // T-100 Varsuk
    "rhsusf_m1a2sep2wd_usarmy",                                                // T-140 Angara
    "rhsusf_m1a2sep2wd_usarmy",                                               // T-140K Angara
    "RHS_UH1Y_FFAR",                                 // Po-30 Orca (Armed)
    "RHS_UH1Y_FFAR",                                 // Po-30 Orca (Armed)
    "RHS_UH60M2",                                      // Mi-290 Taru (Bench)
    "RHS_AH64D_wd"                                 // Mi-48 Kajman
];

// All enemy vehicles that can spawn as battlegroups, either assaulting or as reinforcements, at lower enemy combat readiness (aggression levels).
opfor_battlegroup_vehicles_low_intensity = [
    "rhsusf_m1240a1_m2_usmc_wd",                                                  // Ifrit (HMG)
    "rhsusf_m1240a1_m2_usmc_wd",                                                  // Ifrit (HMG)
    "rhsusf_M1078A1P2_WD_fmtv_usarmy",                                           // Tempest Transport
    "rhsusf_stryker_m1126_m2_wd",                                          // MSE-3 Marid
    "rhsusf_m113_usarmy",                                        // BTR-K Kamysh
    "RHS_M2A2_wd",                                            // ZSU-39 Tigris
    "RHS_UH1Y_FFAR",                                 // Po-30 Orca (Armed)
    "RHS_UH60M2"                                       // Mi-290 Taru (Bench)
];

/* All vehicles that spawn within battlegroups (see the above 2 arrays) and also hold 8 soldiers as passengers.
If something in this array can't hold all 8 soldiers then buggy behaviours may occur.    */
opfor_troup_transports = [
    "rhsusf_M1078A1P2_WD_fmtv_usarmy",                                           // Tempest Transport
    "rhsusf_M1078A1P2_WD_fmtv_usarmy",                                             // Tempest Transport (Covered)
    "rhsusf_stryker_m1126_m2_wd",                                          // MSE-3 Marid
    "RHS_UH60M2",                                      // Mi-290 Taru (Bench)
    "RHS_UH1Y_d"                                  // Po-30 Orca (Armed)
];

// Enemy rotary-wings that will need to spawn in flight.
opfor_choppers = [
    "RHS_UH60M2",                                      // Mi-290 Taru (Bench)
    "RHS_UH1Y_d",                                 // Po-30 Orca (Armed)
    "RHS_AH64D_wd"                                 // Mi-48 Kajman
];

// Enemy fixed-wings that will need to spawn in the air.
opfor_air = [
    "RHS_A10",                                  // To-199 Neophron (CAS)
    "rhsusf_f22"                                              // To-201 Shikra
];
`
